dbt/adapters/bigquery/relation.py

```diff
@@ -90,8 +90,16 @@ def materialized_view_config_changeset(
         new_materialized_view = cls.materialized_view_from_relation_config(relation_config)
 
         if new_materialized_view.options != existing_materialized_view.options:
+            # allow_non_incremental_definition cannot be changed via ALTER, must recreate
+            if (
```

Comment:
I would swap the order here. I think it reads more naturally in the form of "if this setting changed, then recreate, otherwise alter", similar to your comment above.